packages/react/src/use-action-form/types.ts

```diff
@@ -41,13 +41,21 @@ export type FormInput<InputT, Depth extends number = 9, CurrentDepth extends num
   ? { [K in keyof InputT]: FormInput<InputT[K], Depth, Increment<CurrentDepth>> }
   : InputT | null | undefined;
 
+export type StripNulls<T> = T extends null
+  ? undefined
+  : T extends (infer U)[]
+  ? StripNulls<U>[]
+  : T extends object
+  ? { [K in keyof T]: StripNulls<T[K]> }
+  : T;
+
```

Comment:
nit: Do you think it's worth adding a commented blurb summarizing basically this PR here? I've absolutely made it a habit of checking git blame and tracking down a PR for more info on weird things, but I always super appreciate a quick comment.